test/gmp/AxelarGasService.js

```diff
@@ -299,37 +300,171 @@ describe('AxelarGasService', () => {
                     value: nativeGasFeeAmount,
                 });
 
-            await expect(gasService.connect(userWallet).refund(userWallet.address, ethers.constants.AddressZero, nativeGasFeeAmount)).to.be
-                .reverted;
+            await expect(
+                gasService
+                    .connect(userWallet)
+                    .functions['refund(address,address,uint256)'](userWallet.address, AddressZero, nativeGasFeeAmount),
+            ).to.be.reverted;
 
             await expect(
-                await gasService.connect(ownerWallet).refund(userWallet.address, ethers.constants.AddressZero, nativeGasFeeAmount),
-            ).to.changeEtherBalance(userWallet, nativeGasFeeAmount);
+                gasService
+                    .connect(ownerWallet)
+                    .functions['refund(address,address,uint256)'](userWallet.address, AddressZero, nativeGasFeeAmount),
+            )
+                .to.changeEtherBalance(userWallet, nativeGasFeeAmount)
+                .and.to.emit(gasService, 'Refunded')
+                .withArgs(HashZero, 0, userWallet.address, AddressZero, nativeGasFeeAmount);
 
-            await expect(gasService.connect(userWallet).refund(userWallet.address, testToken.address, gasFeeAmount)).to.be.reverted;
+            await expect(
+                gasService
+                    .connect(userWallet)
+                    .functions['refund(address,address,uint256)'](userWallet.address, testToken.address, gasFeeAmount),
+            ).to.be.reverted;
 
-            await expect(await gasService.connect(ownerWallet).refund(userWallet.address, testToken.address, gasFeeAmount))
+            await expect(
+                gasService
+                    .connect(ownerWallet)
+                    .functions['refund(address,address,uint256)'](userWallet.address, testToken.address, gasFeeAmount),
```

Comment:
```suggestion
                    .refund(userWallet.address, testToken.address, gasFeeAmount),
```
Why not just call it directly everywhere? It'll be more readable

Reply:
nope js doesn't support function overloading, and hardhat isn't smart enough to work with it.

Reply:
I think the contract itself being readable is more important than the js testing. I'd slightly prefer a different name if we plan to keep both functions. However, in this case of one being deprecated and potential removed in the near future, I'd say this ugliness in the js code is a temporary thing anyway.

Reply:
`refundExcessGas` makes sense if the function does the calculation. In our case, it's all up to the collector so the "excess" part is confusing.

Reply:
Yeah, the concern is lower due to that. I do think the name I mentioned above is more readable than just refund, but not a strong opinion.